### PR TITLE
Only hide outermost host nodes when Offscreen is hidden

### PR DESCRIPTION
**Pull request by bvaughn**
Follow-up to https://github.com/facebook/react/pull/21079

If there's a hidden Offscreen (or Suspense) subtree containing nested host nodes, we only need to hide the outermost one, e.g. before this change:
```
<Offscreen hidden>
  <div hidden>
    <div hidden>
    [no text]
```
After this change:
```
<Offscreen hidden>
  <div hidden>
    <div>
    [text]
```

### CODE REVIEW

**Comment by gaearon** [comment]
Not just Offscreen, but Suspense too, right?

**Comment by bvaughn** [comment]
> Not just Offscreen, but Suspense too, right?

Suspense uses Offscreen, so yeah it impacts both :)

**Review by acdlite — APPROVED** [review]

File: packages/react-reconciler/src/ReactFiberCommitWork.new.js [hunk 1]
```diff
@@ -1027,14 +1027,18 @@ function hideOrUnhideAllChildren(finishedWork, isHidden) {
...
+
   if (supportsMutation) {
     // We only have the top Fiber that was inserted but we need to recurse down its
     // children to find all the terminal nodes.
     let node: Fiber = finishedWork;
     while (true) {
       if (node.tag === HostComponent) {
         const instance = node.stateNode;
-        if (isHidden) {
+        if (isHidden && hiddenHostSubtreeRoot === null) {
```
Comment (acdlite) [inline packages/react-reconciler/src/ReactFiberCommitWork.new.js:1040]: Should this be wrapped around the `unhideInstance` branch, too?

Comment (acdlite) [reply packages/react-reconciler/src/ReactFiberCommitWork.new.js:1040]: The inner boundary would be responsible for unhiding its children in that case

**Pull request merged**